lib/v2/screens/app/interactor/usecases/approve_guardian_recovery_use_case.dart

```diff
@@ -1,10 +1,19 @@
 import 'package:async/async.dart';
 import 'package:seeds/v2/datasource/remote/api/guardians_repository.dart';
+import 'package:seeds/v2/datasource/remote/firebase/firebase_database_guardians_repository.dart';
 
 class ApproveGuardianRecoveryUseCase {
   final GuardiansRepository _guardiansRepository = GuardiansRepository();
+  final FirebaseDatabaseGuardiansRepository _databaseGuardiansRepository = FirebaseDatabaseGuardiansRepository();
 
-  Future<Result> approveGuardianRecovery(String userAccount, String publicKey) {
-    return _guardiansRepository.recoverAccount(userAccount, publicKey);
+  Future<Result> approveGuardianRecovery(String userAccount, String publicKey) async {
+    var result = await _guardiansRepository.recoverAccount(userAccount, publicKey);
+
+    // If recover call success, Init the firebase recovery flag.
```

Comment:
if this is not success then is ignored and return to the app?

Reply:
correct.